catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SendEvent.java

```diff
@@ -224,9 +234,28 @@ public void deleted(Metacard oldMetacard) {
 
     }
 
-    public boolean ping() {
-        send(HttpMethod.HEAD, null);
-        return subject != null;
+    @Override
+    public boolean isAvailable() {
+        if (hasExperation(subject)) {
+            if (!Security.getInstance()
+                    .tokenAboutToExpire(subject)) {
+                return true;
+            }
+        } else if (lastSuccessfulPing != null && lastSuccessfulPing.isBefore(LocalDateTime.now()
+                .minus(30, ChronoUnit.MINUTES))) {
+            return true;
+        }
+
+        try {
+            Response response = send(HttpMethod.HEAD, null);
+            //HEAD doesn't seem to work so we will assume that any response is good
```

Comment:
I have it defined but I gave up trying to figure out why it wasn't working I don't necessarily care if I get a 200 Ok response either, since I have the subject from the TLS handshake and if it doesn't throw a host not found exception then everything should be peachy keen.